zerver/tests/test_users.py

```diff
@@ -825,10 +825,10 @@ def test_get_accounts_for_email(self) -> None:
         reset_emails_in_zulip_realm()
 
         def check_account_present_in_accounts(user: UserProfile, accounts: List[Dict[str, Optional[str]]]) -> None:
+            realm = user.realm
             for account in accounts:
-                realm = user.realm
                 if account["avatar"] == avatar_url(user) and account["full_name"] == user.full_name \
-                        and account["realm_name"] == realm.name and account["string_id"] == realm.string_id:
+                        and account["realm_name"] == realm.name and account["string_id"] == realm.string_id and account["is_active"]:
                     return
```

Comment:
I would prefer to do the check with an independent verification -- we make sure there is indeed a deactivated account to find (by creating one), and then confirm that it doesn't appear in the output.  

The change to `get_accounts_for_email` is counterproductive -- we already have assured that all accounts will be active, so adding that field to the dictionary just suggests it's possible for that inactive accounts to be there.

Reply:
Thanks for the explanation,
I have changed this commit.